From 45559a470604bd9494d0832589411fc5b683b316 Mon Sep 17 00:00:00 2001
From: Antonio Rojas <arojas@archlinux.org>
Date: Sat, 27 Oct 2018 13:16:04 +0200
Subject: Fix build with poppler<0.64, take 2

Fix build breakage with poppler<0.64 due to changes in 2993cf64b878b78e6b66fddd80f3cbd3b6c66cde

Differential Revision: https://phabricator.kde.org/D16455
---
 filters/karbon/pdf/SvgOutputDev.cpp | 4 ++++
 filters/karbon/pdf/SvgOutputDev.h   | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/filters/karbon/pdf/SvgOutputDev.cpp b/filters/karbon/pdf/SvgOutputDev.cpp
index 870a940..9063e69 100644
--- a/filters/karbon/pdf/SvgOutputDev.cpp
+++ b/filters/karbon/pdf/SvgOutputDev.cpp
@@ -387,7 +387,11 @@ QString SvgOutputDev::printStroke()
     return stroke;
 }
 
+#ifdef HAVE_POPPLER_PRE_0_64
+void SvgOutputDev::drawString(GfxState * state, GooString * s)
+#else
 void SvgOutputDev::drawString(GfxState * state, const GooString * s)
+#endif
 {
     int render = state->getRender();
     // check for invisible text -- this is used by Acrobat Capture
diff --git a/filters/karbon/pdf/SvgOutputDev.h b/filters/karbon/pdf/SvgOutputDev.h
index df2805c..f0d180f 100644
--- a/filters/karbon/pdf/SvgOutputDev.h
+++ b/filters/karbon/pdf/SvgOutputDev.h
@@ -58,7 +58,11 @@ public:
     void eoFill(GfxState *state) override;
 
     // text
+#ifdef HAVE_POPPLER_PRE_0_64
+    void drawString(GfxState * state, GooString * s) override;
+#else
     void drawString(GfxState * state, const GooString * s) override;
+#endif
 
     // images
     void drawImage(GfxState *state, Object *ref, Stream *str,
-- 
cgit v1.1

